### PR TITLE
Xenos can force marine to go out from cryo, sleeper, autodoc

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title

## Why It's Good For The Game

Prevent marine delaying without admin interaction

## Changelog
:cl:
balance: Xenos can force marine to go out from cryo, sleeper, autodoc
/:cl:

### CODE REVIEW

**Comment by Derrick2Electricboogaloo** [comment]
And god said, let them hide in cryo.

File: code/game/objects/machinery/Sleeper.dm [hunk 1]
```diff
@@ -439,6 +439,15 @@
 	else
 		to_chat(user, span_notice("There is no one inside!"))
 
+/obj/machinery/sleeper/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(!occupant)
+		return
+	if(X.status_flags & INCORPOREAL || X.do_actions)
+		return
+	if(!do_after(X, 2 SECONDS))
```
Comment (hyper2snyper) [inline code/game/objects/machinery/Sleeper.dm:447]: can we combine all of these?

File: code/game/objects/machinery/adv_med.dm [hunk 2]
```diff
@@ -119,6 +119,15 @@
 	for(var/obj/O in src)
 		O.forceMove(loc)
 
+/obj/machinery/bodyscanner/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(!occupant)
+		return
+	if(X.status_flags & INCORPOREAL || X.do_actions)
+		return
+	if(!do_after(X, 2 SECONDS))
```
Comment (hyper2snyper) [inline code/game/objects/machinery/adv_med.dm:127]: same here

File: code/game/objects/machinery/autodoc.dm [hunk 3]
```diff
@@ -173,6 +173,15 @@
 	if(updating_health)
 		occupant.updatehealth()
 
+/obj/machinery/autodoc/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(!occupant)
+		return
+	if(X.status_flags & INCORPOREAL || X.do_actions)
+		return
+	if(!do_after(X, 2 SECONDS))
```
Comment (hyper2snyper) [inline code/game/objects/machinery/autodoc.dm:181]: again here.. this is mostly for my sanity.

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
so just combine those checks.
If you dont want to just ping me and call me a donkey again idc.

File: code/game/objects/machinery/Sleeper.dm (hunk 1 above)
Comment (XSlayer300) [reply code/game/objects/machinery/Sleeper.dm:447]: Do you also think we can have feedback when opening the sleepers?

**Comment by BraveMole** [comment]
> so just combine those checks. If you dont want to just ping me and call me a donkey again idc.

I added feedback

**Pull request merged**